File: app/src/main/java/cheema/hardeep/sahibdeep/studentify/activities/HomeActivity.java
package cheema.hardeep.sahibdeep.studentify.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import cheema.hardeep.sahibdeep.studentify.R;
import cheema.hardeep.sahibdeep.studentify.fragments.ClassesFragment;

public class HomeActivity extends AppCompatActivity implements BottomNavigationView.OnNavigationItemSelectedListener {

    public static Intent createIntent(Context context) {
        return new Intent(context, HomeActivity.class);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);
        getSupportActionBar().hide();
        BottomNavigationView bottomNavigationView = findViewById(R.id.bottomNav);
        bottomNavigationView.setOnNavigationItemSelectedListener(this);
        loadFragment(new ClassesFragment());
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
        switch (menuItem.getItemId()) {
            case R.id.menu_classes:
                loadFragment(new ClassesFragment());
                return true;
            case R.id.menu_schedule:
//                loadFragment(new ScheduleFragment());
                startActivity(ClassInformationActivity.createIntent(this));
                return true;
            case R.id.menu_user_information:
                startActivity(UserInformationActivity.createIntent(this));
                return true;
        }
        return false;
    }

    public void loadFragment(Fragment fragment) {
        FragmentManager manager = getSupportFragmentManager();
        manager.beginTransaction().replace(R.id.fragmentContainer, fragment).commit();
    }

}

File: app/src/main/java/cheema/hardeep/sahibdeep/studentify/activities/HomeActivity.kt
package cheema.hardeep.sahibdeep.studentify.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import cheema.hardeep.sahibdeep.studentify.R
import cheema.hardeep.sahibdeep.studentify.fragments.ClassesFragment
import cheema.hardeep.sahibdeep.studentify.fragments.ScheduleFragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.activity_home.*

fun createHomeActivityIntent(context: Context): Intent =
        Intent(context, HomeActivity::class.java)
                .also { Log.d("Sahib", "Created Home Activity Intent") } // expression function

class HomeActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemSelectedListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)
        supportActionBar?.hide()
        bottomNav?.setOnNavigationItemSelectedListener(this)
        loadFragment(ClassesFragment())
    }

    override fun onNavigationItemSelected(menuItem: MenuItem): Boolean =
        when(menuItem.itemId) {
            R.id.menu_classes -> true.also{ loadFragment(ClassesFragment()) }
            R.id.menu_schedule -> true.also{ loadFragment(ScheduleFragment()) }
            R.id.menu_user_information -> true.also{ startActivity(UserInformationActivity.createIntent(this)) }
            else -> false
        }

    private fun loadFragment(fragment: Fragment) {
        supportFragmentManager.beginTransaction().replace(R.id.fragmentContainer, fragment).commit()
    }
}
